components/ingest-service/server/chef.go

```diff
@@ -527,15 +527,59 @@ func (s *ChefIngestServer) processReindexing(ctx context.Context, destIndex, src
 	}
 
 	// Wait for the reindexing to complete
+	taskCompleted := s.monitorReindexing(ctx, requestID, srcIndex, taskID) // Ensure monitoring is finished
 
-	err = s.db.CreateOrUpdateStageAndStatusForIndex(requestID, originalIndex, stage, STATUS_COMPLETED, time.Now())
-	if err != nil {
-		log.Errorf("Failed to update the status for stage %s with status %s with error %v", stage, STATUS_COMPLETED, err)
-		return err
+	// Only update status to COMPLETED if the task was successful
+	if taskCompleted {
+		err = s.db.CreateOrUpdateStageAndStatusForIndex(requestID, originalIndex, stage, STATUS_COMPLETED, time.Now())
+		if err != nil {
+			log.Errorf("Failed to update status for stage %s to COMPLETED: %v", stage, err)
+			return err
+		}
+	} else {
+		log.Warnf("Reindexing task %s for index %s did not complete successfully. Updating status to FAILED.", taskID, srcIndex)
+		if err := s.db.CreateOrUpdateStageAndStatusForIndex(requestID, originalIndex, stage, STATUS_FAILED, time.Now()); err != nil {
+			log.Errorf("Failed to update status for stage %s to FAILED: %v", stage, err)
+		}
 	}
 	return nil
 }
 
+func (s *ChefIngestServer) monitorReindexing(ctx context.Context, requestID int, index, taskID string) bool {
+	// Set a timeout for how long to wait for the task to complete (e.g., 30 minutes).
+	timeout := time.Now().Add(30 * time.Minute)
```

Comment:
30 minutes is not the ideal time out, as we may get large amount of data in some cases, keep it as 6hours for now.